tests/scripts/thread-cert/Cert_5_3_07_DuplicateAddress.py

```diff
@@ -33,110 +33,195 @@
+        self.nodes[ROUTER2].add_prefix(ON_MESH_PREFIX, 'paros')
         self.nodes[ROUTER2].register_netdata()
 
-        # Set lowpan context of sniffer
-        self.simulator.set_lowpan_context(1, '2001:2:0:1::/64')
-
-        self.nodes[MED1].add_ipaddr('2001:2:0:1::1234')
-        self.nodes[SED1].add_ipaddr('2001:2:0:1::1234')
-
+        self.nodes[MED1].add_ipaddr(IPV6_ADDR)
+        self.nodes[SED1].add_ipaddr(IPV6_ADDR)
         self.simulator.go(5)
 
-        # 4
-        # Flush the message queue to avoid possible impact on follow-up
-        # verification.
-        self.simulator.get_messages_sent_by(DUT_LEADER)
-
-        self.nodes[MED3].ping('2001:2:0:1::1234')
-
-        # Verify DUT_LEADER sent an Address Query Request to the Realm local
-        # address.
-        dut_messages = self.simulator.get_messages_sent_by(DUT_LEADER)
-        msg = dut_messages.next_coap_message('0.02', '/a/aq')
-        command.check_address_query(msg, self.nodes[DUT_LEADER], config.REALM_LOCAL_ALL_ROUTERS_ADDRESS)
-
-        # 5 & 6
-        # Verify DUT_LEADER sent an Address Error Notification to the Realm
-        # local address.
+        self.nodes[MED2].ping(IPV6_ADDR)
         self.simulator.go(5)
-        dut_messages = self.simulator.get_messages_sent_by(DUT_LEADER)
-        msg = dut_messages.next_coap_message('0.02', '/a/ae')
-        command.check_address_error_notification(msg, self.nodes[DUT_LEADER], config.REALM_LOCAL_ALL_ROUTERS_ADDRESS)
+
+    def verify(self, pv):
+        pkts = pv.pkts
+        pv.summary.show()
+
+        LEADER = pv.vars['LEADER']
+        LEADER_RLOC = pv.vars['LEADER_RLOC']
+        ROUTER_1 = pv.vars['ROUTER_1']
+        ROUTER_2 = pv.vars['ROUTER_2']
+        MED_1 = pv.vars['MED_1']
+        MED_2 = pv.vars['MED_2']
+        SED = pv.vars['SED']
+        MM = pv.vars['MM_PORT']
+
+        # Step 1: Ensure topology is formed correctly
+        for i in (1, 2):
+            pv.verify_attached('ROUTER_%d' % i, 'LEADER')
+        pv.verify_attached('MED_1', 'ROUTER_1', 'MTD')
+        pv.verify_attached('SED', 'ROUTER_2', 'MTD')
+        pv.verify_attached('MED_2', 'LEADER', 'MTD')
+
+        # Step 5: MED_2 sends ICMPv6 Echo Request to address configured on MED_1
+        #         and SED_1 with Prefix 2001::
+        #         The DUT MUST multicast an Address Query message to the Realm-Local
+        #         All Routers address (FF03::2):
+        #             CoAP URI-Path
+        #                 - NON POST coap://<FF03::2>
+        #             CoAP Payload
+        #                 - Target EID TLV
+
+        pkts.filter_ping_request().\
+            filter_wpan_src64(MED_2). \
+            filter_ipv6_dst(IPV6_ADDR). \
+            must_next()
+        pkts.filter_wpan_src64(LEADER).\
+            filter_ipv6_dst(REALM_LOCAL_ALL_ROUTERS_ADDRESS).\
+            filter_coap_request(ADDR_QRY_URI, port=MM).\
+            filter(lambda p: p.thread_address.tlv.target_eid == Ipv6Addr(IPV6_ADDR)).\
```

Comment:
```suggestion
            filter(lambda p: p.thread_address.tlv.target_eid == IPV6_ADDR).\
```
Casting to `Ipv6Addr` is not required.